native/src/main/java/io/ballerina/stdlib/http/api/service/endpoint/Start.java

```diff
@@ -83,5 +84,15 @@ private static Object startServerConnector(BObject serviceEndpoint) {
         return null;
     }
 
+    private static Runtime getRuntime(Environment env, HTTPServicesRegistry httpServicesRegistry) {
+        if (httpServicesRegistry.getRuntime() != null) {
+            return httpServicesRegistry.getRuntime();
+        } else {
+            Runtime runtime = env.getRuntime();
```

Comment:
Do we have cases where the `httpServicesRegistry.getRuntime() == null` ? We usually set the runtime during the service registry IINM

Reply:
Yes, we set the runtime when we register a service. But when we have a listener and do not register any services on the listener, we returns an error saying `no service has registered for listener` (runtime will be `null` since `register` method is not called). Since we invoke listener level interceptors regardless of the existence of the target service we need a runtime to execute these interceptors

Reply:
Got it